nixos/modules/services/misc/zigbee2mqtt.nix

```diff
@@ -5,26 +5,8 @@ with lib;
 let
   cfg = config.services.zigbee2mqtt;
 
-  configJSON = pkgs.writeText "configuration.json"
-    (builtins.toJSON (recursiveUpdate defaultConfig cfg.config));
-  configFile = pkgs.runCommand "configuration.yaml" { preferLocalBuild = true; } ''
-    ${pkgs.remarshal}/bin/json2yaml -i ${configJSON} -o $out
-  '';
-
-  # the default config contains all required settings,
-  # so the service starts up without crashing.
-  defaultConfig = {
-    homeassistant = false;
-    permit_join = false;
-    mqtt = {
-      base_topic = "zigbee2mqtt";
-      server = "mqtt://localhost:1883";
-    };
-    serial.port = "/dev/ttyACM0";
-    # put device configuration into separate file because configuration.yaml
-    # is copied from the store on startup
-    devices = "devices.yaml";
```

Comment:
@mweinelt 
Did you intentionally leave out the devices.yaml-configuration?
When I leave that out, zigbee2mqtt puts the device config in configuration.yaml, which is overwritten every start.

Apart from that, your PR works fine now.

Reply:
Odd, I wrote a comment in the new preset to that regard, not sure where it went missing. Will check up asap.

Reply:
Fixed.

Reply:
@mweinelt 
Works fine, now :+1: 
Really nice that the test continues to work.

How did you determine the systemd settings for hardening?
Did you use some kind of tool, or did you work it out manually?